codegen_glibmm/dbustypes.py

```diff
@@ -22,108 +22,247 @@
-        """
-        The returned tuple has the following values, in order:
-            - Type for "in"-parameter to generated function
-            - Type for "out" parameter to generated function
-            - Type for use with D-Bus function
-            - function for casting D-Bus type to out-type
-            - function for casting out-type to D-Bus type
-        """
-        if sig == 'b':
-            return ('bool', 'bool', 'bool', "", "")
-        elif sig == 'y':
-            return ('guchar', 'guchar', 'guchar', "", "")
-        elif sig == 'n':
-            return ('gint16', 'gint16', 'gint16', "", "")
-        elif sig == 'q':
-            return ('guint16', 'guint16', 'guint16', "", "")
-        elif sig == 'i':
-            return ('gint32', 'gint32', 'gint32', "", "")
-        elif sig == 'u':
-            return ('guint32', 'guint32', 'guint32', "", "")
-        elif sig == 'x':
-            return ('gint64', 'gint64', 'gint64', "", "")
-        elif sig == 't':
-            return ('guint64', 'guint64', 'guint64', "", "")
-        elif sig == 'd':
-            return ('double', 'double', 'double', "", "")
-        elif sig == 's':
-            return ('std::string', 'std::string', 'Glib::ustring', "Glib::ustring", "")
-        elif sig == 'o':
-            return ('std::string', 'std::string', 'Glib::ustring', "", "")
-        elif sig == 'g':
-            return ('std::string', 'std::string', 'Glib::ustring', "", "")
-        elif sig == 'ay':
-            return ('std::string', 'std::string', 'std::string', "", "")
-        elif sig == 'as':
-            return ('std::vector<std::string> ', 'std::vector<std::string>', 'std::vector<Glib::ustring>', "TypeWrap::glibStringVecToStdStringVec", "TypeWrap::stdStringVecToGlibStringVec")
-        elif sig == 'ao':
-            return ('std::vector<std::string> ', 'std::vector<std::string>', 'std::vector<std::string>', "", "")
-        elif sig == 'aay':
-            return ('std::vector<std::string> ', 'std::vector<std::string>', 'std::vector<std::string>', "", "")
-        elif sig == 'v':
-            return ('Glib::VariantBase', 'Glib::VariantBase', '', '', '')
-        else:
-            return (None, None, None, None, None)
-
 class Annotation:
     def __init__(self, key, value):
         self.key = key
         self.value = value
         self.annotations = []
 
-class Arg:
-    def __init__(self, name, signature):
-        self.name = name
+class Type:
+    def __init__(self, signature, cpptype = ''):
         self.signature = signature
-        self.annotations = []
+        self.cpptype = cpptype
+
+        # Used when:
+        # - passing value from client to proxy methods
+        # - passing value from client to proxy property setters
+        # - in stub property setters, called after getting the value from
+        #   D-Bus and before invoking the property handler
+        # - as input parameter for the stub property handlers
+        #self.cpptype_in
```

Comment:
It's a bit tricky: it's done for documentation purposes only, but we cannot uncomment this line and assign `None` to it, otherwise, the `__getattr__()` method will not be called for this member (`__getatt__()` is only called when the desired member is not present in the class, see [its documentation](https://docs.python.org/3/reference/datamodel.html#object.__getattr__)).